From c3c792bd3f9f4718f500c456a3835e4c5823adae Mon Sep 17 00:00:00 2001
From: Ariel Aviad <aaviad@habana.ai>
Date: Wed, 19 Mar 2025 13:14:32 +0200
Subject: [PATCH] [SW-222736] host: packages: add hl littlefs generate

add a new package for hl-littlefs-generate app to generate
littlefs flash image

Change-Id: Ic8cb7878b51a73491276b19f0a71afd29520c09e
Signed-off-by: Ariel Aviad <aaviad@habana.ai>
---

diff --git a/Config.in.host b/Config.in.host
index d280629..b5cc018 100644
--- a/Config.in.host
+++ b/Config.in.host
@@ -7,4 +7,5 @@
 	source "$BR2_EXTERNAL_HABANALABS_PATH/package/hl-test/Config.in.host"
 	source "$BR2_EXTERNAL_HABANALABS_PATH/package/hl-i2c-tools/Config.in.host"
 	source "$BR2_EXTERNAL_HABANALABS_PATH/package/hlmi/Config.in.host"
+	source "$BR2_EXTERNAL_HABANALABS_PATH/package/hl-littlefs-generate/Config.in.host"
 endmenu
diff --git a/board/habanalabs/common/local.mk b/board/habanalabs/common/local.mk
index a8d7981..239ba8e 100644
--- a/board/habanalabs/common/local.mk
+++ b/board/habanalabs/common/local.mk
@@ -29,3 +29,4 @@
 AGENT_SERDES_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-plugins
 PREBOOT_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-apps/preboot
 PREBOOT_D1_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-apps/preboot/app/preboot_d1
+HL_LITTLEFS_GENERATE_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-modules/littlefs/generate-image
diff --git a/configs/hl_host_defconfig b/configs/hl_host_defconfig
index ac99bde..30b0b43 100644
--- a/configs/hl_host_defconfig
+++ b/configs/hl_host_defconfig
@@ -12,3 +12,4 @@
 BR2_PACKAGE_HOST_DTC=y
 BR2_PACKAGE_HOST_HL_FW_LOADER=y
 BR2_PACKAGE_HOST_HL_SMI=y
+BR2_PACKAGE_HOST_HL_LITTLEFS_GENERATE=y
diff --git a/package/hl-littlefs-generate/Config.in.host b/package/hl-littlefs-generate/Config.in.host
new file mode 100644
index 0000000..011df96
--- /dev/null
+++ b/package/hl-littlefs-generate/Config.in.host
@@ -0,0 +1,6 @@
+config BR2_PACKAGE_HOST_HL_LITTLEFS_GENERATE
+	bool "Habanalabs littleFS generate"
+	default n
+	help
+	  Habanalabs littleFS generate is use to create littleFS
+	  file system image to store at the flash.
diff --git a/package/hl-littlefs-generate/hl-littlefs-generate.mk b/package/hl-littlefs-generate/hl-littlefs-generate.mk
new file mode 100644
index 0000000..1b38f50
--- /dev/null
+++ b/package/hl-littlefs-generate/hl-littlefs-generate.mk
@@ -0,0 +1,18 @@
+HL_LITTLEFS_GENERATE_SITE = hl-littlefs-generate
+HL_LITTLEFS_GENERATE_LICENSE = unclear
+
+DEVICE_NAME = $(firstword $(subst _, ,$(HL_DEV)))
+
+HOST_HL_LITTLEFS_GENERATE_DEPENDENCIES =
+
+HOST_HL_LITTLEFS_GENERATE_CONF_OPTS  = -DTESTS_ENABLED=OFF
+HOST_HL_LITTLEFS_GENERATE_CONF_OPTS += -DCMAKE_EXE_LINKER_FLAGS="-static"
+HOST_HL_LITTLEFS_GENERATE_CONF_OPTS += -DDEVICE_NAME=$(DEVICE_NAME)
+HOST_HL_LITTLEFS_GENERATE_CONF_OPTS += -DCMAKE_BUILD_TYPE="Release"
+HOST_HL_LITTLEFS_GENERATE_CONF_OPTS += -DEMBEDDED_SRC=$(dir $(BR2_EXTERNAL_HABANALABS_PATH))
+
+define HOST_HL_LITTLEFS_GENERATE_INSTALL_CMDS
+	$(INSTALL) -s -D -m 0755 $(@D)/bin/hl-littlefs-generate $(BINARIES_DIR)
+endef
+
+$(eval $(host-cmake-package))
